src/main/java/com/hedera/hashgraph/sdk/account/AccountId.java

```diff
@@ -25,16 +26,14 @@ public AccountId(long shardNum, long realmNum, long accountNum) {
 
     /** Constructs an `AccountId` from a string formatted as <shardNum>.<realmNum>.<accountNum> */
     public static AccountId fromString(String account) throws IllegalArgumentException {
-        var rawNums = Splitter.on('.')
-            .split(account)
-            .iterator();
 
+        long[] rawNums = IdUtil.parseIdString(account);
         var newAccount = AccountID.newBuilder();
 
         try {
-            newAccount.setRealmNum(Integer.parseInt(rawNums.next()));
-            newAccount.setShardNum(Integer.parseInt(rawNums.next()));
-            newAccount.setAccountNum(Integer.parseInt(rawNums.next()));
+            newAccount.setRealmNum(rawNums[0]);
+            newAccount.setShardNum(rawNums[1]);
+            newAccount.setAccountNum(rawNums[2]);
```

Comment:
We have a `AccountId(long, long, long)` constructor, you don't need to call this yourself.

Reply:
The builder calls were in the wrong order originally anyway; I believe it's supposed to go `shard, realm, account/contract/file` but I'm not sure. It hasn't mattered yet because both have been `0` so far. It's probably been confused in other places as well. @gregscullard, @lbaird, can you comment on the intended ordering here?